pkg/operator/controller/ingress/controller.go

```diff
@@ -822,10 +859,23 @@ func (r *reconciler) ensureIngressController(ci *operatorv1.IngressController, d
 		errs = append(errs, fmt.Errorf("failed to list pods in namespace %q: %v", operatorcontroller.DefaultOperatorNamespace, err))
 	}
 
-	errs = append(errs, r.syncIngressControllerStatus(ci, deployment, deploymentRef, pods.Items, lbService, operandEvents.Items, wildcardRecord, dnsConfig, infraConfig))
+	if syncStatusErr, updated := r.syncIngressControllerStatus(ci, deployment, deploymentRef, pods.Items, lbService, operandEvents.Items, wildcardRecord, dnsConfig, infraConfig); syncStatusErr != nil {
+		errs = append(errs, syncStatusErr)
+		// If syncIngressControllerStatus updated our ingress status, it's important we query for that new object.
+		// If we don't, then the next function syncRouteStatus would always fail because it has a stale ingress object.
+		if updated {
+			updatedIc := &operatorv1.IngressController{}
+			if err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: ci.Namespace, Name: ci.Name}, updatedIc); err != nil {
+				errs = append(errs, fmt.Errorf("failed to get ingresscontroller: %w", err))
+			}
+			ci = updatedIc
+		}
+	}
 
 	SetIngressControllerNLBMetric(ci)
 
+	errs = append(errs, r.syncRouteStatus(ci)...)
```

Comment:
Do you think I should do the same pattern above? Since I potentially update the ingress controller object here, subsequent updates after this function (in the coding of the future) will fail. Or is that something we can just leave to future developers?